src/components/PointerCapture/PointerCapture.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+import { MOUSE_PRIMARY } from '../../constants';
+
+export enum Status {
+    dragging = 'dragging',
+    drawing = 'drawing',
+    init = 'init',
+}
+
+type Props = {
+    children: React.ReactNode;
+    className: string;
+    onDrawStart: (x: number, y: number) => void;
+    onDrawStop: () => void;
+    onDrawUpdate: (x: number, y: number) => void;
+    onMouseOut?: () => void;
+    onMouseOver?: () => void;
+    status: Status;
```

Comment:
Can this value be managed internally with an `onStatusChange` callback to inform the parent? Or does the state need to be read at render time in the parent components?

Reply:
That seems possible, I initially had the `status` set differently in the `DrawingCreator` but they have since converged, so maybe the status can be managed internally. The state is used at render time in the parent components, but it'd likely be set back to state based on the `onStatusChange` callback and trigger a re-render also.

Reply:
I tried it but it introduced a lag (and a few other issues as well) so will keep it as is